pyracing/client.py

```diff
@@ -42,6 +42,31 @@ def __init__(self, username: str, password: str):
         self.password = password
         self.session = httpx.AsyncClient()
 
+    def _rename_numerical_keys(self, response_item, mapping):
+        """ Returns a version of 'response_item' where all numerical keys have 
+        been renamed to readable strings by applying 'mapping'.
+        """
+        if not isinstance(response_item, dict):
+            logger.warning('The response_item you provided does not appear to be a '
+                'dictionary. Check if you are calling the '
+                '_rename_numerical_keys method correctly.')
+        
+        if not isinstance(mapping, dict):
+            logger.warning('The mapping you provided does not appear to be a '
+                'dictionary. Check if you are calling the '
+                '_rename_numerical_keys method correctly.')
+        
+        for numerical_key in list(mapping):
```

Comment:
We just checked that mapping is a dict above, why are we converting it to a list? Are we going over the keys or the values here?

Maybe we should just have the method that uses this method pass in the mapping in the format we actually want to use.

Reply:
We are going over the keys in the mapping, check the response_item for a corresponding value and if there is one, assign it to the readable_key...